texar/modules/encoders/xlnet_encoder.py

```diff
@@ -0,0 +1,582 @@
+                A value of 1 for ``target_mapping[i, j, k]`` indicates that
+                the `i`-th target token (in order of permutation) in batch `k`
+                is the token at position `j`.
+                Each row ``target_mapping[i, :, k]`` can have no more than one
+                value of 1.
+            bi_data (bool): Whether to use bidirectional data input pipeline.
+            clamp_len (int): Clamp all relative distances larger than
+                :attr:`clamp_len`. A value of -1 means no clamping.
+            cache_len (int): Length of memory (number of tokens) to cache.
+            same_length (bool): Whether to use the same attention length for
+                each token.
+            attn_type (str): Attention type. Supported values are `"uni"`
+                and `"bi"`.
+            two_stream (bool): Whether to use two-stream attention. Only set to
+                `True` when pre-training or generating text. Defaults to
+                `False`.
+
+        :returns: A tuple of `(output, new_memory)`:
+
+            - **`output`**: The final layer output representations. Shape
+              `[batch_size, seq_len, hidden_dim]`.
+            - **`new_memory`**: The memory of the current batch.
+              If `cache_len` is 0, then `new_memory` is `None`. Otherwise, it is
+              a list of length `num_layers`, each tensor of shape
+              `[batch_size, cache_len, hidden_dim]`.
+              This can be used as the :attr:`memory` argument in the next batch.
+        """
+        return self._forward(self.word_embed(token_ids),
+                             segment_ids=segment_ids,
+                             input_mask=input_mask,
+                             memory=memory,
+                             permute_mask=permute_mask,
+                             target_mapping=target_mapping,
+                             bi_data=bi_data,
+                             clamp_len=clamp_len,
+                             cache_len=cache_len,
+                             same_length=same_length,
+                             attn_type=attn_type,
+                             two_stream=two_stream)
+
+    def _forward(self,
+                 word_embed: torch.Tensor,
+                 segment_ids: Optional[torch.LongTensor] = None,
+                 input_mask: Optional[torch.Tensor] = None,
+                 memory: Optional[List[torch.Tensor]] = None,
+                 permute_mask: Optional[torch.Tensor] = None,
+                 target_mapping: Optional[torch.Tensor] = None,
+                 bi_data: bool = False,
+                 clamp_len: Optional[int] = None,
+                 cache_len: int = 0,
+                 same_length: bool = False,
+                 attn_type: str = 'bi',
+                 two_stream: bool = False) \
+            -> Tuple[torch.Tensor, Optional[List[torch.Tensor]]]:
+        r"""Compute XLNet representations for the input. This layer exists
```

Comment:
If the docstring is the same as that of `forward`, we can simply say this and omit the details. Or, if it's mostly the same, just point out the differences